### ISSUE TITLE
Singularity panics when no GOPATH and HOME are set

### ISSUE THREAD

**Issue by sashayakovtseva** [apptainer/singularity#3163]
GOPATH is not set by default and equals to $HOME/go. When trying to call singularity from CRI without HOME set panic occurs. 

## Version of Singularity:

Any version with plugins.

### Expected behavior

```
sasha:singularity$  HOME="" singularity 
Usage:
  singularity [global options...] <command>

Available Commands:
  apps        List available apps within a container
  build       Build a Singularity image
  cache       Manage the local cache
  capability  Manage Linux capabilities for users and groups
  exec        Run a command within a container
  inspect     Show metadata for an image
  instance    Manage containers running as services
  key         Manage OpenPGP keys
  oci         Manage OCI containers
  plugin      Manage singularity plugins
  pull        Pull an image from a URI
  push        Upload image to the provided library (default is "https://library.sylabs.io")
  remote      Manage Sylabs Cloud endpoints
  run         Run the user-defined default command within a container
  run-help    Show the help for an image
  search      Search a Library for images
  shell       Run a shell within a container
  sign        Attach a cryptographic signature to an image
  test        Run the user-defined tests within a container
  verify      Verify cryptographic signatures attached to an image
  version     Show the version for Singularity
```

### Actual behavior

```
sasha:singularity$ HOME="" singularity 
panic: runtime error: index out of range

goroutine 1 [running]:
github.com/sylabs/singularity/internal/app/singularity.init.ializers()
	singularity/internal/app/singularity/plugin_compile_linux.go:22 +0x1fe
```

### Steps to reproduce behavior

Unset HOME and call singularity as described above.

**Comment by sashayakovtseva** [comment]
@bauerm97  Is it fixed already?

**Comment by hans** [comment]
FWIW, I just ran into this bug with Singularity version `3.1.0-314.gfc944fc` and can reproduce by unsetting `$HOME`:

```
panic: runtime error: index out of range

goroutine 1 [running]:
github.com/sylabs/singularity/internal/app/singularity.init()
        singularity/internal/app/singularity/plugin_compile_linux.go:22 +0x37f
```

Trying to update my Singularity now, since it looks like the referenced `.go` file was changed before the 3.2 release.

**Comment by hans** [comment]
Not an issue in 3.2.0 anymore. Sorry for the noise!

**Comment by sashayakovtseva** [comment]
Fixed in #3258
